contracts/betMain.sol

```diff
@@ -0,0 +1,279 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.10;
+import "@openzeppelin/contracts/utils/Counters.sol";
+import {Base64} from "./libraries/Base64.sol";
+
+contract betMain {
+    using Counters for Counters.Counter;
+    Counters.Counter public _betIDs;
+    Counters.Counter public match_id;
+    uint256 baseBetValue = 10 ether;
+
+    //mapping (address => uint) payabletoContract;
+    address ownerOne = 0x65eb6D9b9Ff3F9999aF0cE71c6331A16Fc19f4b1;
+
+    mapping(uint256 => MatchStruct) matches;
+
+    struct MatchStruct {
+        uint256 _matchId;
+        string teamOne;
+        string teamTwo;
+        string winningTeam;
+        uint256[] betIds;
+    }
+    //Mapping matchId to MatchStruct
+    mapping(uint256 => MatchStruct) matchIdToMatchStruct;
+
+    event MatchUpdateLauncher(
+        uint256 matchId,
+        string teamOne,
+        string teamTwo,
+        string winningTeam,
+        uint256[] betIds
+    );
+    // Need to set this up
+    uint256 currentUserWinAmount;
+
+    struct BetStruct {
+        uint256 matchId;
+        uint256 betId;
+        uint256 betAmount;
+        address betMakerAddress;
+        address betTakerAddress;
+        address betWinnerAddress;
+        string makerBetTeam;
+    }
+
+    event BetEventLauncher(
+        uint256 matchId,
+        uint256 betId,
+        uint256 betAmount,
+        address betMakerAddress,
+        address betTakerAddress,
+        address betWinnerAddress,
+        string makerBetTeam
+    );
+
+    //BetStruct[] betsArray;
+    //betId and BetStruct mapping
+    mapping(uint256 => BetStruct) betIdToStruct;
+
+    //winner address to Clamable amount
+    mapping(address => uint256) betWinnerAmountClaimable;
+
+    modifier onlyOwner() {
+        require(ownerOne == msg.sender, "Only Owners modifier");
+        _;
+    }
+
+    //@dev  This fucntion can be called by anyone having balance > 10 Matic
+    //
+    function _createBet(
+        uint256 betAmount,
+        uint256 _matchId,
+        string memory _TeamName
+    ) external payable {
+        //require(msg.sender.balance >= baseBetValue && msg.sender.balance > betAmount, "Please check the wallet balance");
+
+        // msg.value
+        uint256 existingBalance = betWinnerAmountClaimable[msg.sender];
+
+        if (existingBalance == 0) {
```

Comment:
if there is no existing user, then it'll always come here right.
So, for all new users.